### PR TITLE
Issue 11760 fix tool strip combo box display

### LINKED ISSUE #11760: Weird display when hovering to ToolStripComboBox drop down arrow
### .NET version

9.0.100-preview.7.24371.4 with dlls built from winforms repo of main branch. It cannot be reproduced with .NET SDK.

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes, this is a regression issue. Not reproduce on .NET 8.0

### Issue description

When hovering over drop-down arrow of ToolStripComboBox, an additional blue hover effect appears outside the icon. This issue can only be reproduced with ToolStripComboBox which is in ContextMenuStrip, ToolStrip, MenuStrip. The issue cannot be reproduced with normal ComboBox.

https://github.com/user-attachments/assets/65127428-e7df-4c92-878d-def027ca0be6

### Steps to reproduce

1. Create a WinForms .NET 9.0 project.
2. Add a ToolStrip component to the form.
3. Click the drop-down arrow on the ToolStrip to open the list of available items.
4. Choose ComboBox from the list to add it to the ToolStrip.
5. Build and run project.
6. Hover to drop-down arrow of the ToolStripComboBox and observe the result.

### PR DESCRIPTION
**Pull request by LeafShi1**
Fixes #11760

## Proposed changes

- When the control is ToolStripComboBox, redraw the control when the mouse hovers. (It's mean that revert the changes of PR #11529 when the ComboBox's dropdown button is using small button)

## Customer Impact

- The ToolStripComboBox can be used normally without weird display

## Regression? 

- Yes

## Risk

- Minimal

### Before
When hovering over drop-down arrow of ToolStripComboBox, an additional blue hover effect appears outside the icon.

https://github.com/user-attachments/assets/65127428-e7df-4c92-878d-def027ca0be6

### After

No additional blue hover effect appears outside the icon.

KeepSmallButton

## Test methodology 

- Manually

## Test environment(s) 

- 9.0.0-preview.7.24370.5

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11761)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/ComboBox/ComboBox.FlatComboAdapter.cs [hunk 1]
```diff
@@ -23,6 +23,8 @@ internal class FlatComboAdapter
         private const int OFFSET_2PIXELS = 2;
         protected static int s_offsetPixels = OFFSET_2PIXELS;
 
+        private bool SmallButton { get; }
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/ComboBox/ComboBox.FlatComboAdapter.cs:26]: What does small button refer to here? Is it whether a specific small button is present on the combobox or not?

Comment (LeafShi1) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/ComboBox/ComboBox.FlatComboAdapter.cs:26]: > What does small button refer to here? Is it whether a specific small button is present on the combobox or not?

Yes, It refers to whether to redraw the original DropDown button into a small button. ToolStripComboBox needs to be narrower than the regular ComboBox.

Comment (lonitra) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/ComboBox/ComboBox.FlatComboAdapter.cs:26]: I see, could we add a comment about this and rename this to be more descriptive? Perhaps `ShouldRedrawAsSmallButton` or something similar

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/ComboBox/ComboBox.FlatComboAdapter.cs [hunk 2]
```diff
@@ -46,7 +50,7 @@ public FlatComboAdapter(ComboBox comboBox, bool smallButton)
 
             _origRightToLeft = comboBox.RightToLeft;
 
-            if (_origRightToLeft == RightToLeft.Yes)
+            if (smallButton && _origRightToLeft == RightToLeft.Yes)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/ComboBox/ComboBox.FlatComboAdapter.cs:53]: Why is this change needed?

Comment (LeafShi1) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/ComboBox/ComboBox.FlatComboAdapter.cs:53]: > Why is this change needed?

Thanks for pointing that out, it was a redundant conditional and I have removed it.

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!
